src/main/java/data_access/CachingRateMyProfDataAccess.java

```diff
@@ -0,0 +1,38 @@
+package data_access;
+
+import entity.Professor;
+import use_case.ratemyprof.RateMyProfDataAccessInterface;
+
+import java.util.*;
+
+/**
+ * This class caches RateMyProf calls to improve performance and lessen the load on underlying data source.
+ */
+public class CachingRateMyProfDataAccess implements RateMyProfDataAccessInterface {
+    private final RateMyProfDataAccessObject innerFetcher;
+    private final HashMap<String, Professor> cache;
```

Comment:
```suggestion
    private final Map<String, Professor> cache;
```
Maybe define it as a Map rather than HashMap to program to the interface and make the implementation easier to change if needed.